src/NativeScript/ObjC/Unmanaged/UnmanagedPrototype.mm

```diff
@@ -0,0 +1,50 @@
+#include "UnmanagedPrototype.h"
+#include "UnmanagedInstance.h"
+#include "ObjCTypes.h"
+#include "ObjCConstructorBase.h"
+
+namespace NativeScript {
+static char consumedUnmanagedCheck = 'k';
+
+using namespace JSC;
+
+const ClassInfo UnmanagedPrototype::s_info = { "Unmanaged", &Base::s_info, 0, CREATE_METHOD_TABLE(UnmanagedPrototype) };
+
+static EncodedJSValue JSC_HOST_CALL takeRetainedValue(ExecState* execState) {
+    UnmanagedInstance* instance = jsDynamicCast<UnmanagedInstance*>(execState->thisValue());
+    if (instance->data() == &consumedUnmanagedCheck) {
+        return throwVMTypeError(execState, WTF::ASCIILiteral("Unmanaged value has already been consumed."));
+    }
+
+    id result = static_cast<id>(instance->data());
+    instance->setData(&consumedUnmanagedCheck);
+
+    ObjCConstructorBase* returnType = jsCast<ObjCConstructorBase*>(instance->returnType());
+    JSValue retainedValue = NativeScript::toValue(execState, result, returnType->klass());
+
+    [result release];
+
+    return JSValue::encode(retainedValue);
+}
+
+static EncodedJSValue JSC_HOST_CALL takeUnretainedValue(ExecState* execState) {
```

Comment:
`takeRetainedValue` and `takeUnretainedValue` methods differ in one line only (https://github.com/NativeScript/ios-runtime/pull/386/files#diff-d4cef6d3713c4250ebb83cd1a1d4ac25R25). What do you think about extracting the common logic in another method and calling it twice with different arguments? Something like this:
```
static EncodedJSValue takeValue(ExecState* execState, bool retained) {
    ...
    if(retained) {
        [result release];
    }
    ...
}

static EncodedJSValue JSC_HOST_CALL takeRetainedValue(ExecState* execState) {
    return takeValue(execState, true);
}

static EncodedJSValue JSC_HOST_CALL takeUnretainedValue(ExecState* execState) {
    return takeValue(execState, false);
}
```